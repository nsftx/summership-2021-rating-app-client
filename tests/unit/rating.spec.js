
import { mount } from '@vue/test-utils'
import Rating from '../../src/views/rating'
describe('Rating', () => {
  const wrapper = mount(Rating, {
    propsData: {
      settings: {
        numOfEmoticons: 5,
        timeout: 5,
        msg: 'Thank you for your rating.'
      }
    }
  })
  wrapper.setData({
    voted: false
  })
  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  it('should have changeVoted method', function () {
    expect(typeof wrapper.vm.changeVoted).toBe('function')
  })
  wrapper.vm.changeVoted()
  it('should change voted to true from false', function () {
    expect(wrapper.vm.voted).toBe(true)
  })
})
