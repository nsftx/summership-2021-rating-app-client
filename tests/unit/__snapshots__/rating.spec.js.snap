// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating renders correctly 1`] = `
<div
  class="rating-wrapper"
>
  <div
    style="display: none;"
  >
    <img
      alt="thumbs-up"
      class="like-img"
      src="../assets/positive-vote.svg"
    />
     
    <p>
      Rate our service!
    </p>
     
    <div
      class="emoticon-wrapper"
    />
  </div>
   
  <div
    style=""
  >
    <img
      alt="checkmark"
      class="check-img"
      src="../assets/check.svg"
    />
     
    <p>
      Thank you for your rating.
    </p>
  </div>
</div>
`;
